routes/auth/otp.ts

```diff
@@ -0,0 +1,186 @@
+import { Request } from "express";
+import { Response } from "express-serve-static-core";
+import { ParsedQs } from "qs";
+import {
+	AuthMethodType,
+	AuthMethodVerifyRegistrationResponse,
+	AuthMethodVerifyToFetchResponse,
+	OTPAuthVerifyRegistrationRequest,
+	OtpVerificationPayload,
+} from "../../models";
+import { getPKPsForAuthMethod, mintPKP } from "../../lit";
+import fetch from "node-fetch";
+import { utils } from "ethers";
+import { toUtf8Bytes } from "ethers/lib/utils";
+
+// TODO: UPDATE TO DEPLOYED DOMAIN
+const AUTH_SERVER_URL =
+	process.env.AUTH_SERVER_URL || "http://127.0.0.1:8080/api/otp/verify";
+
+async function verifyOtpJWT(jwt: string): Promise<OtpVerificationPayload> {
+	const res = await fetch(AUTH_SERVER_URL, {
+		redirect: "error",
+		method: "POST",
+		headers: {
+			"Content-Type": "application/json",
+		},
+		body: JSON.stringify({
+			token: jwt,
+		}),
+	});
+	if (res.status < 200 || res.status > 299) {
+		throw new Error("Error while verifying token on remote endpoint");
+	}
+	const respBody = await res.json();
+
+	return respBody as OtpVerificationPayload;
+}
+
+export async function otpVerifyToMintHandler(
+	req: Request<
+		{},
+		AuthMethodVerifyRegistrationResponse,
+		OTPAuthVerifyRegistrationRequest,
+		ParsedQs,
+		Record<string, any>
+	>,
+	res: Response<
+		AuthMethodVerifyRegistrationResponse,
+		Record<string, any>,
+		number
+	>,
+) {
+	const { accessToken } = req.body;
+	let payload: OtpVerificationPayload | null;
+	
+	const tmpToken = (" " + accessToken).slice(1);
+	let userId;
+	let tokenBody: Record<string, unknown>;
+	let orgId;
+	try {
+		tokenBody = parseJWT(tmpToken);
+		orgId = (tokenBody.orgId as string).toLowerCase();
+		let message: string = tokenBody['extraData'] as string;
+		let contents = message.split("|");
+
+		if (contents.length !== 2) {
+			throw new Error("invalid message format in token message");
+		}
+
+		userId = contents[0];
+
+		payload = await verifyOtpJWT(accessToken);
+		if (payload.userId !== userId) {
+			throw new Error("UserId does not match token contents");
+		}
+		console.info("Sucessful verification of OTP token", {
+			userid: payload.userId,
+		});
+	} catch (e) {
+		console.error("unable to verify OTP token ", e);
+		return res.status(400).json({
+			error: "Unable to verify OTP token",
+		});
+	}
+
+	// mint PKP for user
+	try {
+		const authMethodId = utils.keccak256(toUtf8Bytes(`${userId}.${orgId}`));
```

Comment:
might want to consider using colons as the the other auth methods use colons as separators - but not a big deal

Reply:
nice catch 🔥